File: server/src/main/java/br/com/fastgondolas/fastos/server/model/Anexo.java
package br.com.fastgondolas.fastos.server.model;

import br.com.fastgondolas.fastos.server.model.AnexoTipo;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "anexos")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Anexo {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String urlArquivo;

    @Enumerated(EnumType.STRING)
    private AnexoTipo tipoArquivo;

    private String descricao;

    @Builder.Default
    private Instant criadoEm = Instant.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ordem_de_servico_id", nullable = false)
    private OrdemDeServico ordemDeServico;
}

File: mobile/src/db/models/Anexo.ts
import { Model, Relation } from '@nozbe/watermelondb';
import {
  date,
  field,
  readonly,
  relation,
  text,
} from '@nozbe/watermelondb/decorators';
import OrdemDeServico from './OrdemDeServico';

export default class Anexo extends Model {
  static table = 'anexos';

  static associations = {
    ordens_de_servico: { type: 'belongs_to', key: 'ordem_de_servico_id' },
  } as const;

  @text('url_arquivo') urlArquivo!: string;
  @text('tipo_arquivo') tipoArquivo!: string;
  @text('descricao') descricao?: string;

  @field('ordem_de_servico_id') ordemDeServicoId!: string;

  @relation('ordens_de_servico', 'ordem_de_servico_id')
  ordemDeServico!: Relation<OrdemDeServico>;

  @readonly @date('created_at') createdAt!: Date;
  @readonly @date('updated_at') updatedAt!: Date;
}
